common/services/allocation.js

```diff
@@ -140,34 +99,68 @@ class AllocationService extends BaseService {
     status,
     sortBy,
     sortDirection,
+    page,
   } = {}) {
     const [moveDateFrom, moveDateTo] = moveDate
 
-    return this.getAll({
+    return this.get({
       isAggregation,
       include: ['from_location', 'to_location'],
+      page,
       // TODO: This can be removed once move count and progress are returned
       // by the API as resouce meta
       includeCancelled: status ? status.includes('cancelled') : false,
-      filter: pickBy({
-        'filter[status]': status,
-        'filter[from_locations]': fromLocations.join(','),
-        'filter[to_locations]': toLocations.join(','),
-        'filter[locations]': locations.join(','),
-        'filter[date_from]': moveDateFrom,
-        'filter[date_to]': moveDateTo,
-        'sort[by]': sortBy,
-        'sort[direction]': sortDirection,
-      }),
+      filter: {
```

Comment:
Does this mean that other code that uses the `filter[property]` pattern should be updated to use this approach once this is merged in?

Reply:
Only if it uses the new `POST` endpoints within the API. If you're calling a `GET` endpoint they should ideally still follow this structure.

But this is to solve a bigger long term issue of the frontend doing the location permissions...

Reply:
This now introduces 2 different ways of using filters with the services, a useful future cleanup would be to bring them in alignment, no matter what backend implementation is being used.

Reply:
So, I don't think it's about the difference in the service. This new implementation just has to adjust how this service interacts with the API client layer.

Previously we were able to use query params to filter results through a `GET` (or the `findAll` method in devour). The problem we encountered with this was we would hit the querystring limit when passing lots of UUIDs into particular filters like locations.

So, for now, the way we've had to implement this is to create a new `POST` endpoint in the API so that we are not restricted by character limit. But in order to use the new endpoint we needed to update the way in which the service interacts with the devour and the API. @cwrw added a [wiki file](https://dsdmoj.atlassian.net/wiki/spaces/PECS4/pages/2777907242/POST+Moves+and+Allocations+Endpoints+for+large+query+strings) on why we've had to do this.

The `filter[property]` pattern is what is documented in the [JSON:API documentation](https://jsonapi.org/format/#fetching-filtering) and is what we should use by default. 

But this change is trying to help solve the wider issue of the API not knowing about a user and their caseload.